remoteappmanager/application.py

```diff
@@ -149,6 +149,16 @@ def start(self):
 
         tornado.ioloop.IOLoop.current().start()
 
+    def urlpath_for_object(self, object):
```

Comment:
Level of abstraction.
The point is this: we have applications, and they are mapped to urls in our server. Applications can't know what is their url, but everything that handles them must know this information.

Who knows this information (better, who is authoritative for this information)? The application. So you should ask the application. What is the url for a given application? and the application replies. Whatever it replies it's added to the reverse proxy.